### ISSUE TITLE
mypy default does not detect missing function arguments, only works with --strict

### ISSUE THREAD

**Issue by hvnsweeting** [python/mypy#4666]
* Are you reporting a bug, or opening a feature request?
Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python 
# argtest.py
def foo(str1, str2):
    # type: (str, str) -> int
    return 1


def bar():
    return 2 + foo()
```

* What is the actual behavior/output?
Only with `--strict` finds the missing arguments, which actually helped by `--check-untyped-defs`

```bash
$ mypy argtest.py # no output
$ mypy --strict argtest.py
argtest.py:6: error: Function is missing a type annotation
argtest.py:7: error: Too few arguments for "foo"
```

* What is the behavior/output you expect?

Default mypy will detect the error, too. 
As new user trying mypy, gradually moving to annotating all functions,
it is hard to find `--check-untyped-defs`. It helps catching errors when I add new argument to my annotated function but forgot to add new argument on callers - which were not annotated yet.
Or if there is other reason to not make it default, we should update the doc in `common issues` suggest users to use this as they are slowly moving to mypy.

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

```
$ mypy --version
mypy 0.560
```
and master
```
$ mypy --version
mypy 0.570-dev-21dcd8081bd404b18305796651ecd2d6dd8950ef
```

**Comment by JelleZijlstra** [comment]
This is (as you imply) expected behavior: mypy does not check unannotated functions by default. This is sensible behavior when one is gradually introducing typing to a large existing codebase, but I agree it can be confusing for people trying out mypy on small code samples.

**Comment by gvanrossum** [comment]
While we could keep this open as a usability issue, in that case I'd rather have a fresh issue that tackles the desired feature head on: enable --check-untyped-defs by default. Though that's going to be a tricky transition.
